lib/index.mjs

```diff
@@ -0,0 +1,35 @@
+import { isTest } from "std-env";
+
+let createRequire = () => {
+  throw new Error(
+    "createRequire is not supported in browser or test environment.",
+  );
+};
+
+try {
+  createRequire = await import("module").then((m) => m.createRequire);
+} catch {}
+
+export default function (filename, opts) {
+  const isBrowser =
+    typeof window !== "undefined" &&
+    typeof document !== "undefined" &&
+    typeof navigator !== "undefined";
+
+  // In browsers, or in Vitest,
+  // we bypass jiti transform and fallback to the native import and let the runtime handles it
+  if (isBrowser || isTest) {
```

Comment:
We cannot always assume this for browser envs. In an environment like jsdom, isBrowser can be true but also jiti could be working.

Reply:
Yeah true, but I am not sure if there is a good way to distinguish jsdom and real browser. Any ideas?